### PR TITLE
Adding -lp and -ss parameters

### PR DESCRIPTION
**Pull request by intelmark**
Adding support for -lp and -ss, removing roundrobin parameters.  Funcitonality matching to current implementations in SVT-AV1

Note: Linux build was reporting a conflicting error between log2f64 function in EbUtility.h and <math.h> function with the same name, which seemed unrelated to the code changes being made (details below).  Renamed the EbUtility.h function to log2f64adapation and build was successful.

[ 96%] Building C object Source/Lib/Codec/CMakeFiles/SvtVp9Enc.dir/EbSourceBasedOperationsProcess.c.o
In file included from /home/mark/SVT-VP9/Source/Lib/Codec/EbMotionEstimation.c:21:0:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:133:1: error: conflicting types for ‘log2f64’
 __MATHCALL (log2,, (_Mdouble_ __x));
 ^
In file included from /home/mark/SVT-VP9/Source/Lib/Codec/EbRateControlTables.h:11:0,
                 from /home/mark/SVT-VP9/Source/Lib/Codec/EbPictureControlSet.h:22,
                 from /home/mark/SVT-VP9/Source/Lib/Codec/EbMotionEstimation.c:11:
/home/mark/SVT-VP9/Source/Lib/Codec/EbUtility.h:97:17: note: previous declaration of ‘log2f64’ was here
 extern uint64_t log2f64(uint64_t x);

### CODE REVIEW

File: Source/App/EbAppConfig.c [hunk 1]
```diff
@@ -343,9 +342,10 @@ void eb_config_ctor(EbConfig *config_ptr)
 
     // ASM Type
     config_ptr->asm_type                                         = 1;
+
     config_ptr->stop_encoder                                     = EB_FALSE;
-    config_ptr->use_round_robin_thread_assignment                = EB_FALSE;
-    config_ptr->target_socket                                    = 1;
+    config_ptr->target_socket                                    = -1;
+	config_ptr->logical_processors                                = 0;
```
Comment (tianjunwork) [inline Source/App/EbAppConfig.c]: fix the indent here?

File: Source/App/EbAppConfig.h [hunk 2]
```diff
@@ -291,8 +291,8 @@ typedef struct EbConfig
     ****************************************/
     uint32_t        channel_id;
     uint32_t        active_channel_count;
-    uint8_t         use_round_robin_thread_assignment;
-    uint8_t         target_socket;
+	int32_t         target_socket;
```
Comment (tianjunwork) [inline Source/App/EbAppConfig.h]: indent

File: Source/App/EbAppContext.c [hunk 3]
```diff
@@ -184,13 +184,14 @@ EbErrorType  copy_configuration_parameters(
...
     callback_data->eb_enc_parameters.active_channel_count = config->active_channel_count;
-    callback_data->eb_enc_parameters.use_round_robin_thread_assignment = (uint8_t)config->use_round_robin_thread_assignment;
     callback_data->eb_enc_parameters.encoder_bit_depth = config->encoder_bit_depth;
     callback_data->eb_enc_parameters.profile = config->profile;
     callback_data->eb_enc_parameters.level = config->level;
     callback_data->eb_enc_parameters.injector_frame_rate = config->injector_frame_rate;
     callback_data->eb_enc_parameters.speed_control_flag = config->speed_control_flag;
     callback_data->eb_enc_parameters.asm_type = config->asm_type;
+	callback_data->eb_enc_parameters.logical_processors = config->logical_processors;
+	callback_data->eb_enc_parameters.target_socket = config->target_socket;
```
Comment (tianjunwork) [inline Source/App/EbAppContext.c]: indent

File: Source/Lib/Codec/EbUtility.c [hunk 4]
```diff
@@ -128,7 +128,7 @@ PaBlockStats  * pa_get_block_stats(int block_index)
  *  Leading Zeros (NLZ) algorithm to get
  *  the log2f of a 64-bit number
  *****************************************/
-inline uint64_t log2f64(uint64_t x)
+inline uint64_t log2f64adaption(uint64_t x)
 {
```
Comment (tianjunwork) [inline Source/Lib/Codec/EbUtility.c]: why the change here?

**Comment by tianjunwork** [comment]
Hi @intelmark, LGTM, please fix the indent of the PR.

File: Source/Lib/Codec/EbUtility.c (hunk 4 above)
Comment (intelmark) [reply Source/Lib/Codec/EbUtility.c]: Note: Linux build was reporting a conflicting error between log2f64 function in EbUtility.h and <math.h> function with the same name, which seemed unrelated to the code changes being made (details below). Renamed the EbUtility.h function to log2f64adapation and build was successful.

Comment (tianjunwork) [reply Source/Lib/Codec/EbUtility.c]: How it is related to adding -lp and -ss logic?

File: Source/Lib/Codec/EbThreads.h [hunk 5]
```diff
@@ -100,6 +100,8 @@ extern    EB_BOOL           alternate_groups;
 #define _GNU_SOURCE
 #include <sched.h>
 #include <pthread.h>
+#undef _GNU_SOURCE
+#undef __USE_GNU
 extern    cpu_set_t                   group_affinity;
```
Comment (tianjunwork) [inline Source/Lib/Codec/EbThreads.h]: I don't understand the change here. Could you pls explain?

Comment (intelmark) [reply Source/Lib/Codec/EbThreads.h]: There is a subtle difference with the AV1 project that this work is adapted from.

In the AVI project, EbMotionEstimation.c has a #include<math.h>, but in HEVC project, EbMotionEstimation.c does not.  This difference is causing the log2f64 naming conflict when _GNU_SOURCE is defined.

An alternative workaround would be to edit the EbMotionEstimation.c to remove the <math.h> include and replace with: extern double floor(double x).  Then the use of _GNU_SOURCE and <sched.h> in EbThreads.h would be similar to that in AV1.
